tests/integration_tests/test_lightgbm.py

```diff
@@ -31,51 +41,58 @@ def test_lightgbm_pruning_callback_call():
         pruning_callback(env)
 
 
-def test_lightgbm_pruning_callback():
-    # type: () -> None
+@pytest.mark.parametrize('cv', CV_FLAGS)
+def test_lightgbm_pruning_callback(cv):
+    # type: (bool) -> None
 
     study = optuna.create_study(pruner=DeterministicPruner(True))
-    study.optimize(objective, n_trials=1)
+    study.optimize(partial(objective, cv=cv), n_trials=1)
+    print(study.trials[0])
     assert study.trials[0].state == optuna.structs.TrialState.PRUNED
 
     study = optuna.create_study(pruner=DeterministicPruner(False))
-    study.optimize(objective, n_trials=1)
+    study.optimize(partial(objective, cv=cv), n_trials=1)
     assert study.trials[0].state == optuna.structs.TrialState.COMPLETE
     assert study.trials[0].value == 1.
 
     # Use non default validation name.
     custom_valid_name = 'my_validation'
     study = optuna.create_study(pruner=DeterministicPruner(False))
-    study.optimize(lambda trial: objective(trial, valid_name=custom_valid_name), n_trials=1)
+    study.optimize(lambda trial: objective(trial, valid_name=custom_valid_name, cv=cv),
+                   n_trials=1)
     assert study.trials[0].state == optuna.structs.TrialState.COMPLETE
     assert study.trials[0].value == 1.
 
 
-def test_lightgbm_pruning_callback_errors():
-    # type: () -> None
+@pytest.mark.parametrize('cv', CV_FLAGS)
+def test_lightgbm_pruning_callback_errors(cv):
+    # type: (bool) -> None
 
     # "maximize" direction isn't supported yet.
     study = optuna.create_study(pruner=DeterministicPruner(False))
     with pytest.raises(ValueError):
-        study.optimize(lambda trial: objective(trial, metric='auc'), n_trials=1, catch=())
+        study.optimize(lambda trial: objective(trial, metric='auc', cv=cv), n_trials=1, catch=())
 
     # Unknown metric
     study = optuna.create_study(pruner=DeterministicPruner(False))
     with pytest.raises(ValueError):
-        study.optimize(lambda trial: objective(trial, metric='foo_metric'), n_trials=1, catch=())
+        study.optimize(lambda trial: objective(trial, metric='foo_metric', cv=cv),
+                       n_trials=1, catch=())
 
     # Unknown validation name
     study = optuna.create_study(pruner=DeterministicPruner(False))
```

Comment:
Thank you for your review.
Could you kindly explain why you thought it is meaningless?
What I though is below.
In the test case, 'foo_metric' isn't included in metrics parameters, so `LightGBMPruningCallback` has no criterion. Thus it should raise a ValueError.

Reply:
Sorry for my ambiguous description. The phrase "the test" indicated the code between [L82-L88](https://github.com/pfnet/optuna/blob/fbed3714c73eec804e729be324f16f0fa3d4814e/tests/integration_tests/test_lightgbm.py#L82-L88), not the whole of `test_lightgbm_pruning_callback_errors` function.
At [L86](https://github.com/pfnet/optuna/blob/fbed3714c73eec804e729be324f16f0fa3d4814e/tests/integration_tests/test_lightgbm.py#L86), the objective function does not take `cv` variable, so the default value of `cv` argument is always used (thus, actually, it only tests the case where `cv = False`). Besides, if we would pass `cv` variable to the objective function (e.g., `lambda trial: objective(trial, valid_name='valid_1', force_default_valid_names=True, cv=cv)`), the invocation of `study.optimize(...)` does not raise any error because `valid_name` parameter has no effect when `cv == True`. For these reasons I said "it is meaningless".

Reply:
Thank you for pointing out in detail, and I also have to say sorry that I was referring to  a different part of the code. Now I understand what you mean, and agree with your claim. Thank you!